Sell/Basket/Repository/BasketRepository.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  BasketRepository.swift
+//  RealifeTech
+//
+//  Created by Mickey Lee on 23/06/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import Foundation
+import Apollo
+
+public final class BasketRepository {
+
+    private let graphQLManager: GraphQLManageable
+
+    init(graphQLManager: GraphQLManageable) {
+        self.graphQLManager = graphQLManager
+    }
+
+    private func transformErrorIfNeccessary(_ errors: [GraphQLError]?) -> BasketError? {
+        guard let errors = errors else { return nil }
+        for error in errors {
+            switch error.extensions?["code"] as? String {
+            case "SELL_BASKET_OUT_OF_STOCK":
+                return .outOfStock
+            case "SELL_BASKET_PRICE_CHANGE":
+                return .priceChange
+            case "SELL_BASKET_MIXED":
+                return .mixedBasket
+            case "SELL_BASKET_EMPTY":
+                return .emptyBasket
+            default:
+                continue
+            }
+        }
+        return .regularErrors(errors)
```

Comment:
Returning the array of 'regular' errors isn't something we usually expose, it seems odd to do it just for the one repository. The other option is to just return nil if we don't match a specific error, and do like other repos - try and unwrap the data and return `.noDataError` if we can't.

I would recommend returning nil unless there's some new policy/pattern for GraphQL error handling with means we now expose the errors array as standard.

- [ ] Consider returning nil if there's no specific error match

Reply:
Thanks for the suggestion and it makes more sense. However, I would like to keep the `regularErrors` case but tweak the type to `regularError(Error)` as the regular error is still emitted on the normal failure case. Therefore, we can reuse the `GraphQLManagerError.noDataError`